### PR TITLE
WL-4148 Adding isValid Domain method

### PR DESCRIPTION
**Pull request by surya-janani**

### CODE REVIEW

File: kernel-util/src/main/java/org/sakaiproject/util/Validator.java [hunk 1]
```diff
@@ -359,7 +359,27 @@ public static String limitFormattedText(String value, int length)
         return ret.toString();
     }
 
- 
+	/**
+	Checks whether the email address has got valid local domain
```
Comment (buckett) [inline kernel-util/src/main/java/org/sakaiproject/util/Validator.java]: What this method does is much more than this. You should explain it's checking if it's an allowed host part. Formatting of JavaDoc is wrong.

File: kernel-util/src/main/java/org/sakaiproject/util/Validator.java [hunk 2]
```diff
@@ -359,7 +359,27 @@ public static String limitFormattedText(String value, int length)
...
+	@param email
+	 */
+	public static boolean isValidDomain(String email) {
```
Comment (buckett) [inline kernel-util/src/main/java/org/sakaiproject/util/Validator.java]: Although the configuration value is called invalid, that to me normally means syntactically valid, whereas this is specifically checking it's it's allowed so maybe `isAllowedEmailDomain(String email)`? Dunno.

**Comment by buckett** [comment]
Good to refactor code out into common method.

File: kernel-util/src/main/java/org/sakaiproject/util/Validator.java (hunk 2 above)
Comment (surya-janani) [inline kernel-util/src/main/java/org/sakaiproject/util/Validator.java]: Can we have it like isAllowedLocalEmailDomain as it checks for official local domains

Comment (buckett) [inline kernel-util/src/main/java/org/sakaiproject/util/Validator.java]: Sure

On 27 October 2015 at 16:16, Surya Janani Govindarajan <
notifications@github.com> wrote:

> In kernel-util/src/main/java/org/sakaiproject/util/Validator.java
> https://github.com/ox-it/wl-kernel/pull/75#discussion_r43145011:
> 
> > @@ -359,7 +359,27 @@ public static String limitFormattedText(String value, int length)
> >          return ret.toString();
> >      }
> > 
> > ## 
> > -   /**
> > -   Checks whether the email address has got valid local domain
> > -   @param email
> > -    */
> > -   public static boolean isValidDomain(String email) {
> 
> Can we have it like isAllowedLocalEmailDomain as it checks for official
> local domains
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ox-it/wl-kernel/pull/75/files#r43145011.

## 

  Matthew Buckett, VLE Developer, IT Services, University of Oxford

**Comment by surya-janani** [comment]
review changes and rebased

**Comment by surya-janani** [comment]
linked with https://github.com/ox-it/wl-site-manage/pull/51

**Comment by nicholaswilson100** [comment]
ok but Ideally we try and avoid static methods as they are are hard to test (not that we test everything anyway!)

**Comment by nicholaswilson100** [comment]
https://dzone.com/articles/why-static-bad-and-how-avoid

**Comment by surya-janani** [comment]
oh.Good to know.

**Pull request merged**